lib/event_sourceable.rb

```diff
@@ -3,4 +3,17 @@
 require "event_sourceable/event_store"
 
 module EventSourceable
-end
\ No newline at end of file
+  def self.metadata
+    Thread.current[:event_sourceable_metadata] ||= {}
```

Comment:
`Thread.current` _may_ not give you the level of protection you want here. You will be fine in instances where the thread is not reused between contexts however, the thread is reused between contexts in environments like Puma or Sidekiq. The `around_action` ensures the resetting between requests for web but you will need something equivalent for sidekiq.  We can build this out more or just use something like:

https://github.com/steveklabnik/request_store
https://github.com/madebylotus/request_store-sidekiq

Reply:
IIRC request store is essentially the Thread.current[] concept we have here but with auto-reset, yeah?

Reply:
yes, it is thread current with a middleware that resets it between requests. The sidekiq gem adds a server middleware that resets between jobs.